tests/mtest/src/main/scala/tests/BaseSuite.scala

```diff
@@ -1,196 +1,59 @@
-    }
-  }
-  def assertEquals[T](obtained: T, expected: T, hint: String = ""): Unit = {
-    if (obtained != expected) {
-      val hintMsg = if (hint.isEmpty) "" else s" (hint: $hint)"
-      assertNoDiff(obtained.toString, expected.toString, hint)
-      fail(s"obtained=<$obtained> != expected=<$expected>$hintMsg")
-    }
-  }
-  def assertNotFile(path: AbsolutePath): Unit = {
-    if (path.isFile) {
-      fail(s"file exists: $path", stackBump = 1)
-    }
-  }
-  def assertIsFile(path: AbsolutePath): Unit = {
-    if (!path.isFile) {
-      fail(s"no such file: $path", stackBump = 1)
-    }
-  }
-  def assertIsNotDirectory(path: AbsolutePath): Unit = {
-    if (path.isDirectory) {
-      fail(s"directory exists: $path", stackBump = 1)
-    }
-  }
-  def assertNoDiff(
-      obtained: String,
-      expected: String,
-      title: String = ""
-  )(implicit filename: sourcecode.File, line: sourcecode.Line): Unit = {
-    DiffAssertions.colored {
-      DiffAssertions.assertNoDiffOrPrintObtained(
-        obtained,
-        expected,
-        title,
-        title
-      )
-    }
-  }
-  override def utestAfterAll(): Unit = afterAll()
-
-  override def utestFormatter: Formatter = new Formatter {
-    override def exceptionMsgColor: Attrs = Attrs.Empty
-    override def exceptionStackFrameHighlighter(
-        s: StackTraceElement
-    ): Boolean = {
-      s.getClassName.startsWith("scala.meta.internal.worksheets.") ||
-      s.getClassName.startsWith("scala.meta.internal.tvp.") ||
-      s.getClassName.startsWith("scala.meta.internal.pc.") ||
-      s.getClassName.startsWith("scala.meta.internal.mtags.") ||
-      s.getClassName.startsWith("scala.meta.internal.metals.") ||
-      s.getClassName.startsWith("scala.meta.metals.") ||
-      s.getClassName.startsWith("scala.meta.pc.") ||
-      (s.getClassName.startsWith("tests") &&
-      !s.getClassName.startsWith("tests.DiffAssertions") &&
-      !s.getClassName.startsWith("tests.MegaSuite"))
-    }
-    override def formatWrapWidth: Int = 3000
-    override def formatException(x: Throwable, leftIndent: String): Str =
-      super.formatException(x, "")
-  }
-  case class FlatTest(name: String, thunk: () => Unit)
-  private val myTests = mutable.ArrayBuffer.empty[FlatTest]
+  def skipSuite: Boolean = false
 
-  def ignore(options: TestOptions)(fun: => Any): Unit = {
-    myTests += FlatTest(
-      utest.ufansi.Color.LightRed(s"IGNORED - ${options.name}").toString(),
-      () => ()
-    )
-  }
-
-  def isTestSuiteEnabled: Boolean = true
-
-  def test(options: TestOptions)(fun: => Any): Unit = {
-    if (isTestSuiteEnabled) {
-      myTests += FlatTest(options.name, () => {
-        if (options.tags.contains(Tag.ExpectFailure)) {
-          intercept[TestFailedException](fun)
-        } else {
-          fun
-        }
-      })
-    }
+  override def funsuiteTests(): Seq[Test] = {
+    if (skipSuite) Seq.empty
+    else super.funsuiteTests()
   }
 
   def testAsync(
```

Comment:
I have considered adding built-in support in `FunSuite` that captures `Future[_]` values and automatically awaits then with a default timeout, which would avoid `testAsync`